tests/virt_autotest/login_console.pm

```diff
@@ -146,6 +146,25 @@ sub login_to_console {
     set_ssh_console_timeout_before_use if (is_remote_backend && is_aarch64 && get_var('IPMI_HW') eq 'thunderx');
     # use console based on ssh to avoid unstable ipmi
     use_ssh_serial_console;
+    # double-check xen role for xen host
+    $self->double_check_xen_role if is_xen_host;
+
+}
+
+#Just only match bootmenu-xen-kernel needle was not enough for xen host if got Xen domain0 kernel panic(bsc#1192258)
+#Need to double-check xen role after matched bootmenu-xen-kernel needle successfully
+sub double_check_xen_role {
+    record_info 'INFO', 'Double-check xen kernel';
+    die 'Double-check xen kernel failed' if (script_run('lsmod | grep xen') != 0);
```

Comment:
Under the current logic of these three checks, given two situations:

- if the host is bootup with xen kernel, then first check is enough, the second and third are redundent; 

- if the host is NOT bootup with xen kernel, the test will die and the second and the third check would not have the chance to be run.

So would it be better if we would change the checking logic to be like:
```
check if xen kernel is bootup
    if yes, then the host is good
    if no, then
       let's locate the reason
          is xen entry present on the grub2?
          is it bootup with a read-only filesystem?
          other reasons you may check, such as is it bootup with kvm and etc?
```

Reply:
Great! , but this `double_check_xen_role()` will deal with the following two situations:
situation 1:
match virttest-bootmenu-xen-kernel needle at login_console step successfully for Xen host, but, next, Xen domain0 occur kernel panic, and then boot up a normal kernel. So, we need to double-check xen kernel here 
situation 2:
unmatch grub2 needle at login_console step, xen host just boot up read-only filesystem at last.
we need to make sure if  read-only filesystem

Reply:
//here 'xen kernel' means 'lsmod | grep xen'. just add these words in my comment above for clearer description.

Reply:
short update, had updated the checking logic as below:
```
check if xen kernel is bootup   //here 'xen kernel' means 'lsmod | grep xen'
    if yes, then the host is good                                                                           ##deal with situation 1
    if no, then
       let's locate the reason
          is xen entry present on the grub2?
     at last, check with Xen VM host filesystem if it was a read-only filesystem   ##deal with situation 2
```

Reply:
checking 'lsmod | grep xen', it is enough to convince it is a fine xen host. it is impossilbe that a host boots up into a XEN  read-only system. So I suggest to move the read-only check to the 'else' statement.

Reply:
sure,  but refer to https://progress.opensuse.org/issues/101812, just only depend on matching needles at login_conole step was not enough for XEN vm host. after double-check xen kernel successfully, we need to check with Xen vm host filesystem too.

Reply:
Rather than 'matching needles', I said 'lsmod | grep xen' is enough. Have we ever experienced the situation that 'lsmod | grep xen' successfully while its filesystem is read-only?

Reply:
IMHO, no yet.  because of we do not check with xen kernel for xen vm host before. 
So, we have no experienced the situation that 'lsmod | grep xen' successfully while its filesystem is read-only. 
But, I had found out the situation that unmatch grub2 needle at login_console step for xen host, AFAIK. next, xen host boot up xen kernel with the read-only filesystem at last, so, we'd better check with xen kernel and make sure if filesystem was read-only together for xen vm host to deal with this situation.

Reply:
Refer to my local testing validation, there was existed one situation that boot up our SLE Xen vm host with xen kernel and read-only filesystem together. So, we'd better check with xen kernel and filesystem together for our Xen host.

Reply:
Given the xen read-only system, we'd better check it your way. but you missed the pure read-only system in your logic, we experienced a few cases that the focus moved to the bottom of the grub menu due to xen-highlighted needle mismatch. Your only check if the xen entry exists in grub2, then the kernel panic and the needle mismatch cases both meet the checking, you can't determine it is kernel panic case or the needle mismatch case, except we don't care which one it is.

Reply:
AFAIK. according to the focus moved to the bottom of the grub2 menu due to xen-highlighted needle mismatch, then, there was only xen kernel + read-only filesystem on our Xen host without the situation that normal kernel +  read-only filesystem.

Reply:
AFAIK. there just only was the xen kernel and read-only filesystem without normal kernel + read-only filesystem after moved to the bottom of the grub2 menu due to xen-highlighted needle mismatch as your mentioned.

Reply:
If that is the case, the checking logic is ok. LGTM.